### ISSUE TITLE
What is the TARGETPLATFORM in the Dockerfile-envoy?

### ISSUE THREAD

**Issue by bgregg** [envoyproxy/envoy#19206]
Hello,

I have managed to successfully compile an Envoy binary, and now I am trying to create a docker container using the new binary (step 2 from this documentation page: https://www.envoyproxy.io/docs/envoy/latest/start/building/local_docker_build ).

However I am getting this error when trying to run the command listed:

`Sending build context to Docker daemon  1.902GB
Step 1/16 : ARG BUILD_OS=ubuntu
Step 2/16 : ARG BUILD_TAG=18.04
Step 3/16 : FROM $BUILD_OS:$BUILD_TAG
 ---> 5a214d77f5d7
Step 4/16 : ARG TARGETPLATFORM
 ---> Using cache
 ---> 49afe34a6df7
Step 5/16 : RUN apt-get update && apt-get upgrade -y     && apt-get install --no-install-recommends -y ca-certificates     && apt-get autoremove -y && apt-get clean     && rm -rf /tmp/* /var/tmp/*     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 0e3fd4a6b269
Step 6/16 : RUN mkdir -p /etc/envoy
 ---> Using cache
 ---> 285469b1634f
Step 7/16 : ARG ENVOY_BINARY=envoy
 ---> Using cache
 ---> 005abb6add82
Step 8/16 : ARG ENVOY_BINARY_SUFFIX=_stripped
 ---> Using cache
 ---> 6fe16925fbd2
Step 9/16 : ADD ${TARGETPLATFORM}/build_${ENVOY_BINARY}_release${ENVOY_BINARY_SUFFIX}/* /usr/local/bin/
ADD failed: no source files were specified`

I noticed in the Dockerfile-envoy that the argument TARGETPLATFORM has no default value, but I am not sure what to set it to (if I should at all). If it helps, I built the binary in a custom directory using this command:

`sudo ENVOY_DOCKER_BUILD_DIR=/mydata/envoy_build_dir ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.release.server_only'`

And I am running a virtualized Ubuntu 18.04 instance.

Any help is appreciated!

**Comment by phlax** [comment]
hi @bgregg 

i dont think it has to do with `TARGETPLATFORM` the problem is that its expecting the binary to be in the default folder, not your custom one

i guess you have a couple of options

- move the binary to the expected folder
- hack the Dockerfile to point to your custom folder

i guess we could make the binary source directory (more) configurable to account for this scenario, but it seems a bit of a corner case

**Comment by phlax** [comment]
> it seems a bit of a corner case

hmm, maybe its not a bad idea

**Comment by phlax** [comment]
apologies - im realising that `TARGETPLATFORM` is the configurable - did you try setting it to your custom folder ?

i was confused by the name of the var and the fact that `$ git grep TARGETPLATFORM` only turns up the dockerfile

**Comment by bgregg** [comment]
Hey @phlax,

Thanks for reaching out. I was finally able to get it to work; I had to run the following command:

`sudo docker build --build-arg TARGETPLATFORM=./linux/amd64 -f ci/Dockerfile-envoy -t envoy .`

I basically searched my local repository for the folders that the Dockerfile was referencing and I found them in ./linux/amd64. Considering this folder is not part of the remote repository, I can only assume that the build process created this directory in my local repo.
